src/core/thread/address_resolver.cpp

```diff
@@ -278,8 +278,8 @@ otError AddressResolver::Resolve(const Ip6::Address &aEid, uint16_t &aRloc16)
         entry->mTimeout = kAddressQueryTimeout;
         entry->mFailures = 0;
         entry->mRetryTimeout = kAddressQueryInitialRetryDelay;
+        SuccessOrExit(error = SendAddressQuery(aEid));
```

Comment:
Suggestion: would be great to update the `AddressResolver::Resolve()` documentation in header file to show that it may return `OT_ERROR_NO_BUFS`...